libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2587,9 +2587,15 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 		))
 		{
 			annotation.isPure = true;
-			m_scope->annotation().contractDependencies.insert(
-				&dynamic_cast<ContractType const&>(*magicType->typeArgument()).contractDefinition()
-			);
+			ContractDefinition const& accessedContract = dynamic_cast<ContractType const&>(*magicType->typeArgument()).contractDefinition();
+			m_scope->annotation().contractDependencies.insert(&accessedContract);
+
+			if (!TypeProvider::contract(accessedContract)->immutableVariables().empty())
```

Comment:
The value you call `contractDefinition()` on above is also the contract type.

Reply:
it's too early, clearly.